### PR TITLE
bug fix #593

### PR DESCRIPTION
**Pull request by wangzhizheng**
bug fix https://github.com/RackHD/RackHD/issues/593

1, check if the exists count reach the MAX user count

2, Auto figure out the empty userID for adding new user
eg:
exists userID 1,2,7.
we will add the new user as id=3

### CODE REVIEW

**Comment by anhou** [comment]
Invite @yyscamper @uppalk1 to review. I think you are familiar with the logic than me in this part.

File: data/templates/set_bmc_credentials.sh [hunk 1]
```diff
@@ -47,14 +49,38 @@ for x in $cmdReturn; do
...
+                        if [ $ID != $i ]; then
+                                newUserNumber=$i
+                                break
+                        fi
+                done
+                if [ $newUserNumber -eq 0 ]; then
+                        newUserNumber=$((userNumber + 1))
+                fi
+        else
+                ecoh 'reach max user count'
```
Comment (yyscamper) [inline data/templates/set_bmc_credentials.sh]: typo:ecoh ==> "echo"

File: data/templates/set_bmc_credentials.sh [hunk 2]
```diff
@@ -47,14 +49,38 @@ for x in $cmdReturn; do
   i=$((i+1))
 done
 
+function get_newUserNumber()
+{
+        cmdReturn=$(sudo ipmitool user summary $channel)
```
Comment (yyscamper) [inline data/templates/set_bmc_credentials.sh]: Please change the indent to the same as before. Or at least to keep the same file use the same indent.

File: data/templates/set_bmc_credentials.sh [hunk 3]
```diff
@@ -47,14 +49,38 @@ for x in $cmdReturn; do
...
+        myarray=(${cmdReturn//$'\n'/ })
+        MaxCount=${myarray[3]}
```
Comment (yyscamper) [inline data/templates/set_bmc_credentials.sh]: Still a naming style suggestion, previously all variable name uses the camel case format (begins with lower letter capital), keep this style will be better. 

The same for one additional variable `ID` below.

File: data/templates/set_bmc_credentials.sh [hunk 4]
```diff
@@ -47,14 +49,38 @@ for x in $cmdReturn; do
...
 
 if [ $check == 0 ]; then
  echo "Creating a new user"
- userNumber=$((userNumber + 1))
- ipmitool user set name $userNumber <%=user%>
- ipmitool user set password $userNumber <%=password%>
- ipmitool channel setaccess $channel $userNumber callin=on ipmi=on link=on privilege=4
- ipmitool user enable $userNumber
+ get_newUserNumber
+# userNumber=$((userNumber + 1))
```
Comment (yyscamper) [inline data/templates/set_bmc_credentials.sh]: Please remove this comment

**Comment by yyscamper** [comment]
@wangzhizheng : Thanks for your contribution. The code logic is good to me, the typo has to be fixed before merging, those style comment is strongly recommended to be fixed as well.

**Comment by yyscamper** [comment]
@jenkins: ok to test

File: data/templates/set_bmc_credentials.sh (hunk 4 above)
Comment (wangzhizheng) [inline data/templates/set_bmc_credentials.sh]: userNumber?
this varible is not be used in next code.

Comment (lanchongyizu) [inline data/templates/set_bmc_credentials.sh]: I think what @yyscamper means is that you need to remove the line 79
`# userNumber=$((userNumber + 1))`

**Comment by lanchongyizu** [comment]
:+1:

**Comment by yyscamper** [comment]
jenkins: test this please

**Pull request merged**